### ISSUE TITLE
[BUG] Opensearch Docker compose container my-app-1 exited (1)

### ISSUE THREAD

**Issue by alexander-schranz** [opensearch-project/OpenSearch#12489]
### Describe the bug

I used opensearch for my search abstraction project [here](https://github.com/schranz-search/schranz-search). Since the release of 2.12.0 opensearch starts to crash without any changes on our side, the docker container just not longer is booting: https://github.com/schranz-search/schranz-search/actions/runs/8040340102/job/22065739391?pr=366 

### Related component

Build

### To Reproduce

1. create a docker-compose.yml

I use a very simple docker compose to run my tests:

```yaml
services:
  opensearch:
    image: opensearchproject/opensearch:2
    environment:
      discovery.type: single-node
      plugins.security.disabled: 'true'
      cluster.routing.allocation.disk.threshold_enabled: 'false'
    ports:
      - "9200:9200"
    healthcheck:
      test: ["CMD-SHELL", "curl --silent --fail localhost:9200/_cluster/health || exit 1"]
      interval: 5s
      timeout: 5s
      retries: 20
    volumes:
      - opensearch-data:/usr/share/opensearch/data

volumes:
  opensearch-data:
```

2. Make sure to have latest opensearch version:

```
docker pull opensearchproject/opensearch:2
```

3. Try to start opensearch

```
docker compose up --wait
```

### Expected behavior

Should work as before.

### Additional Details

**Plugins**

No Plugins

**Screenshots**

![Bildschirmfoto 2024-02-28 um 08 36 34](https://github.com/opensearch-project/OpenSearch/assets/1698337/8eb42486-9830-41ef-8789-c441c991ecbe)

**Host/Environment (please complete the following information):**
 - OS: MacOS (arm) and Ubuntu (Github Action) via Docker 
 - Version latest 2 (2.12.0)

**Additional context**

Sadly I was not yet able to get more output why it even crashes not sure if there are any debug flags I can activate to get here better output.

**Comment by peternied** [comment]
[Triage - attendees [1](https://github.com/peternied) [2](https://github.com/reta) [3](https://github.com/lukas-vlcek/)]
@alexander-schranz thanks for filing.

@DarshitChanpura Looks like this is related to the admin password changes, can you look into this and provide next steps for ?

**Comment by DarshitChanpura** [comment]
@alexander-schranz With 2.12, the security demo configuration setup had a breaking change around the admin credentials. There is no default password "admin" anymore. A custom strong password is required to be passed to successfully setup demo configuration. If none / weak password is provided, the cluster will fail to spin-up.

The issue that you are facing is one of using `plugins.security.disabled` vs `DISABLE_SECURITY_PLUGIN`. (discussed in depth on this issue: https://github.com/opensearch-project/security/issues/4062)

TLDR; `plugins.security.disabled: true` passed via environment variable disables security post demo configuration setup vs using `DISABLE_SECURITY_PLUGIN` which disables security at the time of setup.

Here is a modified docker-compose.yml that would work in your use-case:

```yml
services:
  opensearch:
    image: opensearchproject/opensearch:2
    environment:
      discovery.type: single-node
      cluster.routing.allocation.disk.threshold_enabled: 'false'
      DISABLE_SECURITY_PLUGIN: true
    ports:
      - "9200:9200"
    healthcheck:
      test: ["CMD-SHELL", "curl --silent --fail localhost:9200/_cluster/health || exit 1"]
      interval: 5s
      timeout: 5s
      retries: 20
    volumes:
      - opensearch-data:/usr/share/opensearch/data

volumes:
  opensearch-data:
```

**Comment by alexander-schranz** [comment]
Thx this currently fixes my CI.
